node/commons/compatibility-layer.ts

```diff
@@ -24,71 +18,175 @@ export const convertBiggyProduct = (
       })
     : []
 
-  const skus: any[] = (product.skus || []).map(
+
+  const categoriesIds: string[] = product.categoryIds
+  ? product.categoryIds.map((_: any, index: number) => {
+      const subArray = product.categoryIds.slice(0, index + 1)
+      return `/${subArray.join('/')}/`
+    }).reverse()
+  : []
+
+  const skus: SearchItem[] = (product.skus || []).map(
     convertSKU(product, indexingType, tradePolicy)
   )
 
+  const allSpecifications = product.productSpecifications.concat(getSKUSpecifications(product))
+
+  const allSpecificationsGroups = [ ...product.specificationGroups.keys() ]
+
+  const brandId = product.brandId ? Number(product.brandId) : -1
+
   const selectedProperties = product.split && [
     {
       key: product.split.labelKey,
       value: product.split.labelValue,
     },
   ]
 
-  const convertedProduct: any = {
+  const convertedProduct: SearchProduct & { cacheId?: string, [key: string]: any } = {
     categories,
-    selectedProperties,
+    categoriesIds,
     productId: product.id,
     cacheId: `sp-${product.id}`,
     productName: product.name,
     productReference: product.reference || product.product || product.id,
     linkText: product.link,
-    brand:
-      product.brand ||
-      product.extraInfo['marca'] ||
-      product.extraInfo['brand'] ||
-      '',
-    brandId: -1,
+    brand: product.brand || '',
+    brandId,
     link: product.url,
     description: product.description,
     items: skus,
-    sku: skus.find(sku => sku.sellers && sku.sellers.length > 0),
```

Comment:
why did you remove this?

Reply:
The return in the compatibility layer should be the same return we have in the `fq` API, which is the type SearchProduct (with some other fields that are created on the run, mostly fields that are `categoryName: categoryValues`, this sku field does not exist in there (it's basically the items field)